Anastasiakiseleva patch 1 (#1)
DevExpress-Examples/xaf-how-to-execute-actions-in-code: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E1393)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# XAF - How to execute Actions programmatically

> This example explains how you can trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.

Reviewer: This example shows how to trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.

Author: ```suggestion
This example shows how to trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.
```

> This example explains how to trigger code inside the `Execute` event handlers of custom or standard XAF Actions programmatically, for example, to execute default Action logic that shows a View or runs code in the background.

Reviewer 2: ```suggestion
This example explains how you can trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.
```

> **NOTE** 

> > We do not recommend this approach. Refactor your Action's event handlers and extract required code into separate methods. Call these separate methods directly without triggering UI-related Action code.

Reviewer: We do not recommend this approach. Refactor your Action's event handlers and extract the code into separate methods. Call these separate methods directly without triggering UI-related Action code.

> > We do not recommend this approach. Refactor your Action's `Execute` and other event handlers and extract the required code into separate methods. You can call these methods directly without triggering the UI-related Action code.

Reviewer 2: ```suggestion
> We do not recommend this approach. Refactor your Action's event handlers and extract required code into separate methods. Call these separate methods directly without triggering UI-related Action code.
```

![image](https://user-images.githubusercontent.com/14300209/227978022-a50434fa-413f-4634-b0ba-09936fd82927.png)

## Implementation Details

### SimpleAction

`mySimpleAction.DoExecute();`

### ParametrizedAction

`myParametrizedAction.DoExecute("test value");`

> Mind the `ParametrizedAction.ValueType`. Here, a string is used as an example.

Reviewer 2: ```suggestion
Make sure that the parameter type matches `ParametrizedAction.ValueType`. A string is used in this example.
```

### SingleChoiceAction

`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]);`

> Mind the `SingleChoiceAction.Items` structure. Here, the first item is used as an example.

Reviewer 2: ```suggestion
Make sure the pass a parameter that's a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection.
```

> Make sure the pass a parameter that's a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection.

Reviewer: Make sure you pass a parameter that is a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection
